.github/workflows/model-evaluation.yml

```diff
@@ -0,0 +1,201 @@
+name: Full Model Evaluation
+
+on:
+  workflow_dispatch:
+    inputs:
+      provision-and-deploy:
+        description: 'Whether to provision the deployment'
+        required: true
+        default: true
+        type: boolean
+      deploy-only:
+        description: 'Whether to deploy the application'
+        required: true
+        default: false
+        type: boolean
+      purge:
+        description: 'Whether to purge the deployment after evaluation'
+        required: true
+        default: false
+        type: boolean
+
+permissions:
+  id-token: write
+  contents: read
+
+jobs:
+    deploy-azure:
+      if: ${{ inputs.provision-and-deploy == true || inputs.deploy-only == true }}
+      uses: ./.github/workflows/azure-dev.yml
+      secrets: inherit
+      with:
+        deploy-only: ${{ inputs.deploy-only }}
+
+    evaluate-models:
+        if: ${{ (inputs.provision-and-deploy == false && inputs.deploy-only == false && always()) || success() }}
+        needs: deploy-azure
+        runs-on: ubuntu-latest
+        strategy:
+          matrix:
+            os: ["ubuntu-20.04"]
```

Comment:
Why not a newer one? I see above that the job runs on "ubuntu-latest".

Reply:
my bad, I will remove line 40, that's accidentally introduced and in fact that version value isn't affecting anything.